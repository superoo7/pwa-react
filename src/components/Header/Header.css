.header {
  background: linear-gradient(to right, #24A3CF, #3650A0);
}

.container {
  /* Make the split to edge of each side */
  display: flex;
  justify-content: space-between;
  max-width: var(--site-width);
  margin: 0 auto;
}

.title {
  padding: 1rem 0;
  color: #ffffff;
  font-family: sans-serif;

  font-style: italic;
  font-size: 3.0rem;
  user-select: none;
  cursor: pointer;

}

.list {
  display: flex;
  align-items: center;
  height: 100%;

}

.list-item {
  margin-right: 1rem;
  user-select: none;
  cursor: pointer;
}

.link {
  color: #ffffff;
  font-family: sans-serif;
  text-transform: uppercase;
  font-weight: bold;
  font-size: 2em;
}

.link:hover,
.link:focus {
  color: #cccccc;
}
