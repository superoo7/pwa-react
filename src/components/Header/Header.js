import React from 'react'
import styles from './Header.css'

const Header = () => (
  <header className={styles.header} >
    <div className={styles.container}>
      <h1 className={styles.title}>superoo7</h1>
      <nav>
        <ul className={styles.list}>
          <li className={styles.listItem}>
            <a href='#' className={styles.link}>Posts</a>
          </li>
          <li className={styles.listItem}>
            <a href='#' className={styles.link}>About</a>
          </li>
          <li className={styles.listItem}>
            <a href='#' className={styles.link}>Resume</a>
          </li>
          <li className={styles.listItem}>
            <a href='#' className={styles.link}>Contact</a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
)

export default Header
