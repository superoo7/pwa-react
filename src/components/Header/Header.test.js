import React from 'react'
import { shallow } from 'enzyme'
import Header from './Header'
import toJson from 'enzyme-to-json'


test('Header component should render as expected', () => {
  const component = shallow(<Header />)
  const tree = toJson(component)
  // console.log(tree)
  expect(tree).toMatchSnapshot()
})
