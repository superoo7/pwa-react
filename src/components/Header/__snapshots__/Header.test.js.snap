// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render as expected 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <h1
      className="title"
    >
      superoo7
    </h1>
    <nav>
      <ul
        className="list"
      >
        <li
          className="listItem"
        >
          <Link
            className="link"
            replace={false}
            to="/"
          >
            Posts
          </Link>
        </li>
        <li
          className="listItem"
        >
          <Link
            className="link"
            replace={false}
            to="/about"
          >
            About
          </Link>
        </li>
        <li
          className="listItem"
        >
          <a
            className="link"
            href="#"
          >
            Resume
          </a>
        </li>
        <li
          className="listItem"
        >
          <a
            className="link"
            href="#"
          >
            Contact
          </a>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
